pkg/compact/compact.go

```diff
@@ -473,10 +473,42 @@ func IsHaltError(err error) bool {
 	return ok1 || ok2
 }
 
+func (cg *Group) areBlocksOverlapping(newBlocks ...block.Meta) error {
+	var (
+		metas   []tsdb.BlockMeta
+		exclude = map[ulid.ULID]struct{}{}
+	)
+
+	for _, m := range newBlocks {
+		metas = append(metas, m.BlockMeta)
+
+		for _, s := range m.Compaction.Sources {
+			exclude[s] = struct{}{}
```

Comment:
this is not enough. Still we have blocks that are "within" newBlock but are not in new Block sources ): this is kind of confusing -> we have only leafs there, right?